### PR TITLE
Add support for the DISCO_F429ZI target

### PR DESCRIPTION
**Pull request by AlessandroA**
This PR depends on https://github.com/ARMmbed/mbed-os/pull/3397, so do not merge it yet!

As soon as the mbed OS PR will be merged, I'll also push the updated mbed-os.lib file here.

@meriac @niklas-arm @Patater @adustm

### CODE REVIEW

**Comment by AlessandroA** [comment]
@meriac @niklas-arm PR is no longer WIP and can be reviewed.
**Note: It still can't be merged though, as we need to wait for the mbed OS release.**

**Comment by mazimkhan** [comment]
retest uvisor

**Comment by mazimkhan** [comment]
retest uvisor

File: source/main-hw.h [hunk 1]
```diff
@@ -47,6 +47,32 @@ extern Serial shared_pc;
...
+        {USART1,              sizeof(*USART1), UVISOR_TACLDEF_PERIPH}, \
+        {I2C1,                sizeof(*I2C1),   UVISOR_TACLDEF_PERIPH}, \
+        {SPI1,                sizeof(*SPI1),   UVISOR_TACLDEF_PERIPH}, \
+        {RCC,                 sizeof(*RCC),    UVISOR_TACLDEF_PERIPH}, \
+        {FLASH,               sizeof(*FLASH),  UVISOR_TACLDEF_PERIPH}, \
+        {PWR,                 sizeof(*PWR),    UVISOR_TACLDEF_PERIPH}, \
+        {EXTI,                sizeof(*EXTI),   UVISOR_TACLDEF_PERIPH}, \
+        {GPIOG,               sizeof(*GPIOG),  UVISOR_TACLDEF_PERIPH}, \
+        {SYSCFG,              sizeof(*SYSCFG), UVISOR_TACLDEF_PERIPH}, \
+        {(void *) 0x42000000, 0x01000000,      UVISOR_TACLDEF_PERIPH}, /* FIXME */ \
```
Comment (Patater) [inline source/main-hw.h]: Add a bit more detail to your FIXME comment. It is not clear what you are thinking needs fixing.

**Comment by Patater** [comment]
Don't forget to update mbed-os.lib

**Comment by AlessandroA** [comment]
Closing in favor of #40.

**Pull request closed without merging**